### ISSUE TITLE
[BUG] Error in deserialization of DataFactoryIntegrationRuntimeResource

### ISSUE THREAD

**Issue by Tiktack** [Azure/azure-sdk-for-net#47197]
### Library name and version

Azure.ResourceManager.DataFactory 1.6.0

### Describe the bug

When retrieving DataFactoryIntegrationRuntimeResource from ArmClient it fails on deserializing

System.InvalidOperationException: The requested operation requires an element of type 'Number', but the target element has type 'String'.

<pre><code>System.InvalidOperationException: The requested operation requires an element of type 'Number', but the target element has type 'String'.
   at System.Text.Json.ThrowHelper.ThrowJsonElementWrongTypeException(JsonTokenType expectedType, JsonTokenType actualType)
   at System.Text.Json.JsonDocument.TryGetValue(Int32 index, Int32& value)
   at System.Text.Json.JsonElement.GetInt32()
   at Azure.ResourceManager.DataFactory.Models.IntegrationRuntimeComputeProperties.DeserializeIntegrationRuntimeComputeProperties(JsonElement element, ModelReaderWriterOptions options)
   at Azure.ResourceManager.DataFactory.Models.ManagedIntegrationRuntime.DeserializeManagedIntegrationRuntime(JsonElement element, ModelReaderWriterOptions options)
   at Azure.ResourceManager.DataFactory.Models.DataFactoryIntegrationRuntimeProperties.DeserializeDataFactoryIntegrationRuntimeProperties(JsonElement element, ModelReaderWriterOptions options)
   at Azure.ResourceManager.DataFactory.DataFactoryIntegrationRuntimeData.DeserializeDataFactoryIntegrationRuntimeData(JsonElement element, ModelReaderWriterOptions options)
   at Azure.ResourceManager.DataFactory.IntegrationRuntimesRestOperations.GetAsync(String subscriptionId, String resourceGroupName, String factoryName, String integrationRuntimeName, String ifNoneMatch, CancellationToken cancellationToken)
   at Azure.ResourceManager.DataFactory.DataFactoryIntegrationRuntimeCollection.GetAsync(String integrationRuntimeName, String ifNoneMatch, CancellationToken cancellationToken)</code></pre>

Error happens when trying to get IntegrationRuntime from managed instance ( don't really think that makes a difference)

### Expected behavior

Retrieve DataFactoryIntegrationRuntimeResource for a pipeline or retrieve all pipelines without any problems

### Actual behavior

Exception on deserializing json response

### Reproduction Steps

<pre><code>var credentials = new ClientSecretCredential(TenantId, ApplicationId, AuthenticationKey);
var client = new ArmClient(credentials, SubscriptionId);

ResourceGroupResource resourceGroup = await (await client.GetDefaultSubscriptionAsync())
    .GetResourceGroupAsync(ResourceGroup);

DataFactoryResource dataFactoryResource = await resourceGroup.GetDataFactoryAsync(DataFactoryName);

DataFactoryIntegrationRuntimeResource integrationRuntime = await dataFactoryResource.GetDataFactoryIntegrationRuntimeAsync(IntegrationRuntimeName);
</code></pre>

### Environment

Azure Function V4 with Windows Service Plan. .NET 9

**Comment by Tiktack** [comment]
Hi, any news here? 
This bug prevents our migration from deprecated Microsoft.Management libraries to supported Azure.ResourceManager. 
Right now we found issues only with Azure.ResourceManager.DataFactory

**Comment by ArthurMa1978** [comment]
Thank you @Tiktack for your feedback, @ruowan please look into this issue.

**Comment by wangbwn** [comment]
Hello, @Tiktack . Could you try it locally? I attempted to reproduce it locally but couldn't reproduce the issue.

var cred = new DefaultAzureCredential();
var client = new ArmClient(cred);

DataFactoryResource dataFactoryResource = client.GetDataFactoryResource(new ResourceIdentifier($"/subscriptions/xxxx/resourceGroups/xxxx/providers/Microsoft.DataFactory/factories/xxxxx")).Get();

DataFactoryIntegrationRuntimeResource integrationRuntime = await dataFactoryResource.GetDataFactoryIntegrationRuntimeAsync("integrationRuntime1");

**Comment by Tiktack** [comment]
Hi, the same error. Tried in console app .NET 9 
![Image](https://github.com/user-attachments/assets/deeb79ca-46cb-43ad-88fb-8c688b0974e4)

Here is ADF runtimes: 
![Image](https://github.com/user-attachments/assets/70a32468-f0cb-48c5-bbc3-e3eea3ed5996)

So, I am trying to connect to the IR named "IntegrationRuntime", but with no success.

**Comment by wangbwn** [comment]
Hello, @Tiktack . Sorry, I still couldn’t reproduce the issue. Could you provide more information? For example, the payload of this Azure-SSIS IR.

**Comment by Tiktack** [comment]
@wangbwn 
Is this what you mean? Got it from ADF: 
<pre><code>{
    "name": "IntegrationRuntime",
    "properties": {
        "type": "Managed",
        "typeProperties": {
            "computeProperties": {
                "location": "East US 2",
                "nodeSize": "Standard_D2_v3",
                "numberOfNodes": "1",
                "maxParallelExecutionsPerNode": "4",
                "vNetProperties": {
                    "subnetId": "/subscriptions/XXX/resourceGroups/XXX/providers/Microsoft.Network/virtualNetworks/XXX/subnets/XXX"
                }
            },
            "ssisProperties": {
                "catalogInfo": {
                    "catalogServerEndpoint": "XXX.database.windows.net, 1433",
                    "catalogAdminUserName": "XXX",
                    "catalogAdminPassword": {
                        "type": "SecureString",
                        "value": "**********"
                    }
                },
                "edition": "Standard",
                "licenseType": "LicenseIncluded"
            }
        }
    }
}
</code></pre>

**Comment by wangbwn** [comment]
Yes, thank you!

**Comment by wangbwn** [comment]
Hello, @Tiktack .
![Image](https://github.com/user-attachments/assets/ad9816ad-178e-478c-9dbd-7df9f4602260)
The maxParallelExecutionsPerNode should be of type int. I'm not sure why it appears as a string on your side. Could it be an issue with the environment?

**Comment by Tiktack** [comment]
Also, here is request from dev tools. It uses api version: 2018-06-01 which returns number of nodes and max parallel as strings
![Image](https://github.com/user-attachments/assets/dd209a10-6a5b-4030-be7c-91b3a78d60cc)

**Comment by wangbwn** [comment]
API versions are the same. It should not be possible to create such an IR now. Could you share your creation steps?

**Comment by wangbwn** [comment]
Hello, @Tiktack . If possible, could you follow your creation process to create a new one? Does the same issue occur?

**Comment by Tiktack** [comment]
Hi @wangbwn,
We used to deploy our ADF with IR using ARM templates. Last time we deployed ADF was maybe a year ago with such code
![Image](https://github.com/user-attachments/assets/682ea3c7-b7f1-4900-a77d-768a3623318a)

Both parameters were strings at that time: 
![Image](https://github.com/user-attachments/assets/7a29e2c4-bd78-4610-ad1d-6e5a340911ab)

I will try to change these 2 parameters to numbers and deploy it. 
But it will take sometime, I will reply in a couple of days.
